ui/hooks/experiences/useRamps.test.js

```diff
@@ -1,46 +1,108 @@
-  getCurrentChainId: jest.fn(),
-}));
-
-jest.mock('../../../shared/constants/network', () => ({
-  CHAIN_IDS: {
-    GOERLI: '5',
-    SEPOLIA: '10',
-    MAINNET: '1',
+let mockStoreState = {
+  metamask: {
+    providerConfig: {
+      chainId: '0x1',
+    },
+    metaMetricsId: mockedMetametricsId,
   },
-}));
+};
+
+const wrapper = ({ children }) => (
+  <Provider store={configureStore(mockStoreState)}>{children}</Provider>
+);
 
 describe('useRamps', () => {
-  beforeAll(() => {
+  beforeEach(() => {
+    global.platform = { openTab: jest.fn() };
+  });
+
+  afterEach(() => {
     jest.clearAllMocks();
-    Object.defineProperty(global, 'platform', {
-      value: {
-        openTab: jest.fn(),
+  });
+
+  it('should default the metamask entry param when opening the buy crypto URL', () => {
+    const metaMaskEntry = 'ext_buy_sell_button';
+    const mockChainId = '0x1';
+
+    mockStoreState = {
+      ...mockStoreState,
+      metamask: {
+        ...mockStoreState.metamask,
+        providerConfig: {
+          chainId: mockChainId,
+        },
       },
+    };
+
+    const mockBuyURI = `${process.env.PORTFOLIO_URL}/buy?metamaskEntry=${metaMaskEntry}&chainId=${mockChainId}&metametricsId=${mockedMetametricsId}`;
+    const openTabSpy = jest.spyOn(global.platform, 'openTab');
+
+    const { result } = renderHook(() => useRamps(), { wrapper }); // default metamask entry
+
+    result.current.openBuyCryptoInPdapp();
+    expect(openTabSpy).toHaveBeenCalledWith({
+      url: mockBuyURI,
     });
   });
 
-  it('should open the buy crypto URL for MAINNET chain ID', () => {
-    const mockChainId = '1';
-    const mockBuyURI = `${process.env.PORTFOLIO_URL}/buy?metamaskEntry=ext_buy_sell_button&metametricsId=1`;
+  it('should use the correct metamask entry param when opening the buy crypto URL', () => {
+    const metaMaskEntry = 'ext_buy_banner_tokens';
+    const mockChainId = '0x1';
 
-    useSelector.mockReturnValue(mockChainId);
-    const openTabSpy = jest.spyOn(global.platform, 'openTab');
+    mockStoreState = {
+      ...mockStoreState,
+      metamask: {
+        ...mockStoreState.metamask,
+        providerConfig: {
+          chainId: mockChainId,
+        },
+      },
+    };
 
-    const { result } = renderHook(() => useRamps());
+    const mockBuyURI = `${process.env.PORTFOLIO_URL}/buy?metamaskEntry=${metaMaskEntry}&chainId=${mockChainId}&metametricsId=${mockedMetametricsId}`;
+    const openTabSpy = jest.spyOn(global.platform, 'openTab');
 
-    expect(typeof result.current.openBuyCryptoInPdapp).toBe('function');
+    const { result } = renderHook(
+      () => useRamps(RampsMetaMaskEntry.TokensBanner),
+      { wrapper },
+    );
 
     result.current.openBuyCryptoInPdapp();
-
     expect(openTabSpy).toHaveBeenCalledWith({
       url: mockBuyURI,
     });
   });
+
+  ['0x1', '0x38', '0xa'].forEach((mockChainId) => {
+    it(`should open the buy crypto URL with the currently connected chain ID of ${mockChainId}`, () => {
```

Comment:
nit: use [`it.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) instead

Reply:
[1b42efd](https://github.com/MetaMask/metamask-extension/pull/23352/commits/1b42efdfda01485ca81a5c121f16c4b887198e92)